src/vmm/src/guest_config/common/mod.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO - Remove allow(unused) once types have been used
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::ops::Add;
+
+use serde::{Deserialize, Serialize};
+
+#[cfg(target_arch = "x86_64")]
+use crate::guest_config::cpuid::Cpuid;
+
+/// Type defined to represent registers across CPU vendors
+/// (pointers and values)
+pub trait Numeric: Copy + Add<Self, Output = Self> {}
+impl Numeric for u32 {}
+impl Numeric for u64 {}
+impl Numeric for u128 {}
+
+/// Map to find and define model-specific registers
+#[cfg(target_arch = "x86_64")]
+#[allow(unused)]
+pub struct MsrRegisterMap {
+    register_map: HashMap<RegisterPointer<u32>, RegisterValue<u32>>,
+}
+/// Map to find and define registers for 64-bit ARM CPUs
+#[cfg(target_arch = "aarch64")]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[allow(unused)]
+pub struct Aarch64RegisterMap {
+    register_map: HashMap<RegisterPointer<u64>, RegisterValue<u128>>,
+}
+
+/// A trait type to be customized for each CPU architecture
+/// to modify CPU configuration for the Firecracker guest
+pub trait ConfigurationModifier {
+    /// Applies the CPUTemplate against the CPUConfiguration
+    fn apply_template(&self, cpu_template: CustomCpuTemplate) -> Box<CpuConfiguration>;
+}
+
+/// CPU configuration that can be of the host or the guest.
+/// Expected to be populated in part or modified by
+/// types implementing ConfigurationModifier
+#[allow(unused)]
+pub struct CpuConfiguration {
+    /// Hashed set containing register configuration
+    /// divided up by type of register. See `RegisterSet`
+    pub register_config: Vec<RegisterSet>,
+}
```

Comment:
So from what I can see this struct is not used anywhere. 
But I guess your idea is that it will be used in `vcpu/x86_64` and `vcpu/aarch64` modules.
But then why do we even need this type, if it has totaly different implementations based on the architecture and used only in those implementations. There is nothing in common between x86 and aarch for this struct to exist.

Reply:
It is not supposed to be used anywhere yet. The commonality is that they will be configured by the same API.